### ISSUE TITLE
Request a batch of status messages

### ISSUE THREAD

**Issue by david4096** [ga4gh/workflow-execution-service-schemas#21]
Paraphrased from Bruce Hoff <bruce.hoff@sagebase.org>:

I would like a request that allows me to return a list of status messages for each ID requested, instead of performing get requests for each workflow being tracked.

**Comment by brucehoff** [comment]
This would allow a client to display for the user a *dashboard* of running workflows (retrieving, say, a page at a time rather than making a new request for each row of the table).

**Comment by adamstruck** [comment]
Couldn't this be achieved using the ListWorkflows endpoint with filtering?

**Comment by david4096** [comment]
@adamstruck yes, one way to close this would be to add a parameter `workflow_ids` that accepts an array of strings in the `workflow_list_request` and will only return workflows with the provided identifier in a ListWorkflowResponse.

**Comment by brucehoff** [comment]
> Couldn't this be achieved using the ListWorkflows endpoint with filtering?

Yes, that is a good point.  @david4096 also makes a good suggestion to determine the subset by passing a list of IDs.  This relieves the WES implementation from having to implement filtering, sorting, etc.  Please also see #23 which requests richer status than just a "state" string,  (percent complete or other progress information).  To me ideally the ListWorkflows would accept a list of IDs and return a list of maps (sets of key-value pairs) containing status metadata for the workflow.  It seems to me that we would also need an endpoint to POST status information, that endpoint callable by the workflow itself.

**Comment by david4096** [comment]
Thanks @brucehoff , to actually close this, one would need to get the workflow logs using a filtered list, not just the list of workflows.

**Comment by cjllanwarne** [comment]
Pinging @vsmalladi and @patmagee - this is coming up as a would-be-useful feature for us too (updating CBAS to monitor run statuses in batches). I imagine that having some way to filter results would be useful for anyone who ends up with more than a handful of runs to query against in their engine. Would it be worthwhile me making an implementation PR to modify `ListWorkflows` with filter options as suggested above?

**Comment by vsmalladi** [comment]
@cjllanwarne I think its worth making the PR and then discussing where there might be issues.

**Comment by patmagee** [comment]
@cjllanwarne I agree, if you have the bandwidth could you add a PR for discussion?

**Comment by uniqueg** [comment]
For reference, there is a PR for filtering list outputs in TES as well: https://github.com/ga4gh/task-execution-schemas/pull/170
Requirements and use cases should be almost identical
